util/argo/argo.go

```diff
@@ -123,6 +125,29 @@ func WaitForRefresh(ctx context.Context, appIf v1alpha1.ApplicationInterface, na
 	return nil, fmt.Errorf("application refresh deadline exceeded")
 }
 
+func TestRepo(repo *argoappv1.Repository) error {
+	checks := map[string]func() error{
```

Comment:
do you need to invoke `repo.CopyCredentialsForm`

Reply:
Checked all places where method is used. In all places there is either `repo.CopyCredentialsForm` call or `db.GetRepository` which takes care of credentials copying.